feat: update dotnet version and dependencies

karlos-oliveira opened the pull request:
- updates dotnet version to 8;
- updates dependencies to the last version;
- replaces binaryformatter to JsonSerializer;
- implements some dotnet 8 features where apply.

guilhermeluizsp commented on test/Liquid.Activation.Tests/LightWorkerTests.cs:
```diff
@@ -0,0 +1,154 @@
+        }
+
+        [Fact]
+        public void InvokeProcessWhenMessageIsntValidJsonThrows()
```
By looking at the name of this test method, I thought it would check whether the `InvokeProcess` method would throw an exception if the message parameter is not a valid json. This, however, is not that it is actually doing. Nowhere in its body it's checking if there's an exception being thrown. Should we rename it?

bruno-brant replied:
Hmm, something happened. That _was_ such a test, and there should be a test that covered the null parameters. Anyway, I'm still improving a bit of the coverage, but hope to finish soon so I can start to correct the bugs in this class.

bruno-brant replied:
Done.

bruno-brant commented:
Added a few more tests, now we have good coverage of the code in LightWorker. Hacked around a bit (exposed a few private methods that should be called by derived classes independently, for instance) which helped me achieve a good amount of coverage.

There's a lot of code here that need refactoring and a lot of bad design choices that in later versions we need to remove.

guilhermeluizsp commented on test/Liquid.Activation.Tests/LightWorkerTests.cs:
```diff
@@ -0,0 +1,430 @@
+        [InlineData("anything")]
+        [InlineData(null)]
+        [InlineData(1)]
+        public void InvokeProcessWhenMethodIsNullReturnsExpectedValue(object message)
```
This does say it covers the `InvokeProcess` method when the `method` parameter **is** null, but it is **not** null.

https://github.com/Avanade/Liquid-Application-Framework/blob/4f76069eb609ad20a570af4d44060b1a925782b3/test/Liquid.Activation.Tests/LightWorkerTests.cs#L63

Perhaps I misinterpreted the intention of this test?

guilhermeluizsp commented on test/Liquid.Activation.Tests/LightWorkerTests.cs:
```diff
@@ -0,0 +1,430 @@
+        }
+
+        [Fact]
+        public void InvokeProcessWhenMethodIsEchoMessageIsEmptyReturnsNull()
```
I think there's something wrong with the name of this method: "InvokeProcess, when method is echo message is empty, returns null". Maybe I just didn't get it. Could you enlighten me, please?

Was it supposed to be something like "InvokeProcessWhenMethodIsEchoMethodAndMessageIsEmptyReturnsNull"?

guilhermeluizsp commented on test/Liquid.Activation.Tests/LightWorkerTests.cs:
```diff
@@ -0,0 +1,430 @@
+
+            var message = ToJsonByteStream("anything");
+
+            message[0] = _fixture.Create<byte>();
```
Since `MethodsCollection.EchoMethod` expects an instance of `Foobar` as parameter, this line isn't necessary, because `"anything"` is not a valid serialized `Foobar` already. Is there any reason you're randomizing the first byte?

guilhermeluizsp commented on test/Liquid.Activation.Tests/LightWorkerTests.cs:
```diff
@@ -0,0 +1,430 @@
+
+            var message = ToJsonByteStream("anything");
+
+            message[0] = _fixture.Create<byte>();
```
Kind of the same thing here. Is this randomization really necessary? In this case, you're testing the branch where the method does not have any parameters (line 142):
https://github.com/Avanade/Liquid-Application-Framework/blob/4f76069eb609ad20a570af4d44060b1a925782b3/src/Liquid.Activation/Worker/LightWorker.cs#L134-L150

In this case, we're never deserializing the `message` parameter (`InvokeProcess` will only deserialize when the method has one or more parameters), so it does not make any difference whether it is invalid or not.

guilhermeluizsp replied:
If this happens to be true and the "invalid message creation" part of the code is really not needed, then this test can be safely removed because `InvokeProcessWhenMethodHasZeroParametersDoesntParseMessage` already does this.

guilhermeluizsp commented on test/Liquid.Activation.Tests/LightWorkerTests.cs:
```diff
@@ -0,0 +1,430 @@
+            sut.ValidateInput(viewModel);
+
+            // ACT & ASSERT
+            Assert.ThrowsAny<Exception>(() => sut.Factory<MockDomain>());
```
I was wondering if we could check whether `InvalidInputException` is being thrown here, instead of `Exception`. What do you think?

guilhermeluizsp commented on test/Liquid.Activation.Tests/LightWorkerTests.cs:
```diff
@@ -0,0 +1,430 @@
+
+        // TODO: There are translation errors in the Critics class:(
+        [Fact]
+        public void TerminateWhenCriticsHandlerHasCritics()
```
I think you missed the expected behavior part. Shouldn't it be "TerminateWhenCriticsHandlerHasCriticsThrows"?

Commit:
- update project version to dotnet 8
- update dependencies to the last version
- replace binaryformatter to JsonSerializer
- implement some dotnet 8 features where apply

Commit:
feat: update references

lucianareginalino commented:
Olá @karlos-oliveira , primeiramente muito obrigada pela sua contribuição! 

Estamos trabalhando na release 6.0.0 onde o target framework será o net6.0, por esse motivo peço por favor reabrir seu PR para a branch https://github.com/Avanade/Liquid-Application-Framework/tree/releases/v8.0.0 , sua contribuição será um excelente ponto de partida para a versão 8.0.

Agradeço novamente, e fico no aguardo.

karlos-oliveira commented:
Prontinho @lucianareginalino , abri o [PR](https://github.com/Avanade/Liquid-Application-Framework/pull/211) para a release/v8.0.0, qualquer coisa pode contar comigo!

lucianareginalino commented:
perfeito! obrigada!
